torch/ao/quantization/_pt2e/decompositions.py

```diff
@@ -27,3 +27,56 @@ def dequantize_per_tensor(
     dtype: torch.dtype,
 ) -> torch.Tensor:
     return (input.to(torch.float32) - zero_point) * scale
+
+# Helper function used to implement per-channel quantization against any axis
+def _permute_to_axis_zero(x, axis):
+    new_axis_list = list(range(x.dim()))
+    new_axis_list[axis] = 0
+    new_axis_list[0] = axis
+    y = x.permute(tuple(new_axis_list))
+    return y, new_axis_list
+
+@register_decomposition(torch.ops.quantized_decomposed.quantize_per_channel)
+def quantize_per_channel(
+        input: torch.Tensor,
+        scales: torch.Tensor,
+        zero_points: torch.Tensor,
+        axis: int,
+        quant_min: int,
+        quant_max: int,
+        dtype: torch.dtype
+) -> torch.Tensor:
+    input, permute_axis_list = _permute_to_axis_zero(input, axis)
+    res = torch.zeros_like(input)
+
+    for i in range(input.size(0)):
+        res[i] = torch.clamp(
```

Comment:
could this not be done without the loop if broadcast_right existed: https://github.com/pytorch/pytorch/issues/62148

Reply:
yeah probably, broadcasting is typically implicit though, how do you call broadcast_right?

Reply:
it doesn't exist currently in python :( created a feature request here: https://github.com/pytorch/pytorch/issues/94233

what you could do is sth like `scales = scales.unflatten(-1, scales.shape[-1:] + (1, ) * (res.dim() - scales.dim()))` (though likely not torchscriptable if it matters here)

or `torch.round(input.movedim(0, -1) / scales) + zero_points, quant_min, quant_max).movedim(-1, 0)` - although not sure what it does to preserving memory format

Reply:
this looks a bit complicated, I'll try transpose instead, thanks for the suggestions

Reply:
transpose or movedim is the cleanest, yes